zebrad/tests/acceptance.rs

```diff
@@ -4,195 +4,292 @@
+    let (mut cmd, _guard) = test_cmd(env!("CARGO_BIN_EXE_zebrad"))?;
+
+    cmd.arg(arg)
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .spawn2()
+}
+
+fn get_child_multi_args(args: &[&str]) -> Result<zebra_test::command::TestChild> {
+    let (mut cmd, _guard) = test_cmd(env!("CARGO_BIN_EXE_zebrad"))?;
+
+    cmd.args(args)
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .spawn2()
+}
+
+fn get_child_no_args() -> Result<zebra_test::command::TestChild> {
+    let (mut cmd, _guard) = test_cmd(env!("CARGO_BIN_EXE_zebrad"))?;
+
+    cmd.stdout(Stdio::piped()).stderr(Stdio::piped()).spawn2()
+}
 
 #[test]
-fn generate_no_args() {
-    let mut runner = RUNNER.clone();
-    let mut cmd = runner.arg("generate").capture_stdout().run();
-    cmd.stdout()
-        .expect_line("# Default configuration values for zebrad.");
-    cmd.wait().unwrap().expect_success();
+fn generate_no_args() -> Result<()> {
+    zebra_test::init();
+
+    let child = get_child_single_arg("generate");
+    let output = child.unwrap().wait_with_output()?;
+    let output = output.assert_success()?;
+
+    output.stdout_contains(r"# Default configuration values for zebrad.")?;
+
+    Ok(())
 }
 
 #[test]
-fn generate_args() {
-    let mut runner = RUNNER.clone();
-    let cmd = runner
-        .args(&["generate", "argument"])
-        .capture_stdout()
-        .run();
-    cmd.wait().unwrap().expect_code(1);
+fn generate_args() -> Result<()> {
+    zebra_test::init();
+
+    // Invalid free argument
+    let child = get_child_multi_args(&["generate", "argument"]);
+    let output = child.unwrap().wait_with_output()?;
+    let output = output.assert_failure()?;
+
+    output.stdout_contains(r"unexpected free argument `argument`")?;
 
     // Invalid flag
-    let mut runner = RUNNER.clone();
-    let cmd = runner.args(&["generate", "-f"]).capture_stdout().run();
-    cmd.wait().unwrap().expect_code(1);
+    let child = get_child_multi_args(&["generate", "-f"]);
+    let output = child.unwrap().wait_with_output()?;
+    let output = output.assert_failure()?;
+
+    output.stdout_contains(r"unrecognized option `-f`")?;
 
     // Valid flag but missing argument
-    let mut runner = RUNNER.clone();
-    let cmd = runner.args(&["generate", "-o"]).capture_stdout().run();
-    cmd.wait().unwrap().expect_code(1);
+    let child = get_child_multi_args(&["generate", "-o"]);
+    let output = child.unwrap().wait_with_output()?;
+    let output = output.assert_failure()?;
+
+    output.stdout_contains(r"missing argument to option `-o`")?;
 
-    let mut runner = RUNNER.clone();
-    let cmd = runner
-        .args(&["generate", "-o", "file"])
-        .capture_stdout()
-        .run();
-    cmd.wait().unwrap().expect_success();
+    // Valid
+    let child = get_child_multi_args(&["generate", "-o", "file.yaml"]);
+    let output = child.unwrap().wait_with_output()?;
+    output.assert_failure()?; // should be assert_success?
```

Comment:
probably related to the guard that's being dropped

Reply:
It was, thanks!